### DISCUSSION TITLE
Feature request - add parameters to cm_OpenDirInNewTab

### DISCUSSION THREAD

**Discussion by kgabor545** [doublecmd/doublecmd#1158] [Suggestions]
According to documentation: cm_OpenDirInNewTab "Open directory under cursor at new tab, but don't switch to it."
In reality it do switch to the new tab if I am standing on a dir, and duplicate the tab in background when I am on a file.

The paramaters I requesting are these:
```
side=left/right/opposit
activate=yes/no
```
I thin they are self explanatory.
Other option is to add parameter to cm_TransferLeft and cm_TransferRight
```
newtab=yes/no
activate=yes/no
```
Thank you

**Comment by Skif-off** [comment]
This command has another meaning: `cm_OpenDirInNewTab` works like "Open this link in the background tab" in any internet browser.

Try something like
```lua
local params = {...}

local function OpenDirInNewTab(s)
  DC.ExecuteCommand("cm_NewTab")
  DC.ExecuteCommand("cm_ChangeDir", s)
end

local bO, bA = false, false

if #params == 3 then
  if (params[2] == "opposit") or (params[3] == "opposit") then bO = true end
  if (params[2] == "activate") or (params[3] == "activate") then bA = true end
elseif #params == 2 then
  if params[2] == "opposit" then
    bO = true
  elseif params[2] == "activate" then
    bA = true
  end
elseif #params == 1 then
  DC.ExecuteCommand("cm_OpenDirInNewTab")
else
  return
end

if (bO == true) and (bA = true) then
  DC.ExecuteCommand("cm_FocusSwap")
  OpenDirInNewTab(params[1])
elseif (bO == true) and (bA = false) then
  DC.ExecuteCommand("cm_FocusSwap")
  OpenDirInNewTab(params[1])
  DC.ExecuteCommand("cm_FocusSwap")
elseif (bO == false) and (bA = true) then
  OpenDirInNewTab(params[1])
elseif (bO == false) and (bA = false) then
  DC.ExecuteCommand("cm_OpenDirInNewTab")
end
```
Save as, for example, `OpenDirInNewTab.lua ` (use UTF-8 without BOM), [add button](https://doublecmd.github.io/doc/en/toolbar.html) with internal command `cm_ExecuteScript`, parameters:
```
path-to-OpenDirInNewTab.lua
%"0%p0
opposit
activate
```
`opposit` and `activate ` are optional parameters: if only one parameter (i.e. `%"0%p0`), this script will just call `cm_OpenDirInNewTab`.
If there is a file under the cursor, script will open tab with the same directory and move the cursor to this file: it might be convenient (search results and so on). Change this behavior and work only with directories?
